pyDeltaRCM/sed_tools.py

```diff
@@ -22,275 +24,585 @@ def sed_route(self):
+        self.uy = self._mr.uy  # update component flow field
 
-        Returns
-        -------
+    def topo_diffusion(self):
+        """Diffuse topography after routing.
 
+        Diffuse topography after routing all coarse sediment parcels. The
+        method uses convolution with a kernel to compute smoother topography,
+        and then adds this different to the current eta to do the smoothing.
+        The operation is repeated `N_crossdiff` times.
         """
-        eta_change_loc = Vp_dep / self.dx**2
-
-        self.eta[px, py] = self.eta[px, py] + eta_change_loc  # update bed
-        self.depth[px, py] = self.stage[px, py] - self.eta[px, py]
-
-        if self.depth[px, py] < 0:
-            self.depth[px, py] = 0
+        for _ in range(self.N_crossdiff):
 
-        self.pad_depth[px + 1, py + 1] = self.depth[px, py]
+            a = ndimage.convolve(self.eta, self.kernel1, mode='constant')
+            b = ndimage.convolve(self.qs, self.kernel2, mode='constant')
+            c = ndimage.convolve(self.qs * self.eta, self.kernel2,
+                                 mode='constant')
 
-        if self.depth[px, py] > 0:
-            self.uw[px, py] = min(self.u_max, self.qw[
-                                  px, py] / self.depth[px, py])
+            self.cf = (self.diffusion_multiplier *
+                       (self.qs * a - self.eta * b + c))
 
-        self.update_u(px, py)
+            self.cf[self.cell_type == -2] = 0
+            self.cf[0, :] = 0
 
-        self.Vp_res = self.Vp_res - Vp_dep
-        # update amount of sediment left in parcel
+            self.eta += self.cf
 
-    def erode(self, Vp_ero, px, py):
-        """Erode sand or mud.
 
-        Total sediment mass is preserved but individual categories
-        of sand and mud are not.
+r_spec = [('_dt', float32), ('dx', float32),
+          ('num_starts', int64), ('start_indices', int64[:]),
+          ('stepmax', float32), ('px', int64), ('py', int64),
+          ('eta', float32[:, :]), ('stage', float32[:, :]),
+          ('depth', float32[:, :]), ('cell_type', int64[:, :]),
+          ('uw', float64[:, :]), ('ux', float64[:, :]), ('uy', float64[:, :]),
+          ('pad_stage', float32[:, :]), ('pad_depth', float32[:, :]),
+          ('pad_cell_type', int64[:, :]), ('qw', float64[:, :]),
+          ('qx', float64[:, :]), ('qy', float64[:, :]), ('qs', float64[:, :]),
+          ('ivec_flat', float32[:]), ('jvec_flat', float32[:]),
+          ('iwalk_flat', int64[:]), ('jwalk_flat', int64[:]),
+          ('distances_flat', float32[:]),
+          ('dry_depth', float32), ('gamma', float32), ('_lambda', float32),
+          ('beta', float32),  ('f_bedload', float32),
+          ('theta_sed', float32), ('u_max', float32),
+          ('qs0', float32), ('u0', float32), ('Vp_sed', float32),
+          ('Vp_res', float32), ('Vp_dep_mud', float64[:, :]),
+          ('U_dep_mud', float32), ('U_ero_mud', float32),
+          ('U_ero_sand', float32)]
```

Comment:
I think there's no harm to this. It would certainly make things leaner, but I am not sure by how much. since these parameters are allocated once, and presumably garbage collected after every iteration, I suspect that it wouldn't change much. However, would be excited to be proved wrong!